### ISSUE TITLE
Feature request: use raw text to simplify pseudocode syntax

### ISSUE THREAD

**Issue by OrangeX4** [andreasKroepelin/lovelace#2]
Lovelace is a unopinionated and flexible pseudocode package and i think it is great.

But I still think there's some points for improvement here, why don't we use a simpler raw text to identify Indent? Something like this:

```typ
#import "@preview/lovelace:0.1.0": *

#show: setup-lovelace

#let _typst2lovelace(typst-code, scope: (:)) = {
  let indent = 0
  let last-indent = 0
  let res = ()
  for line in typst-code.text.split("\n") {
    let whitespaces = line.find(regex("^\\s+"))
    let current-indent = if (whitespaces != none) { whitespaces.len() } else { 0 }
    if (indent == 0 and current-indent != 0) {
      indent = current-indent
    }
    if (current-indent > last-indent) {
      res = res + (ind,) * int((current-indent - last-indent) / indent)
    } else if (current-indent < last-indent) {
      res = res + (ded,) * int((last-indent - current-indent) / indent)
    }
    last-indent = current-indent
    let line-code = line.slice(current-indent)
    let match = line-code.match(regex("^<(.*)>\\s*$"))
    if (match != none) {
      res.push(label(match.captures.at(0)))
    } else {
      res.push(eval(line-code, mode: "markup", scope: (no-number: no-number, comment: comment) + scope))
    }
  }
  res
}

#pseudocode(.._typst2lovelace(
  ```typ
  *input:* integers $a$ and $b$
  *output:* greatest common divisor of $a$ and $b$
  *while* $a != b$ *do*
    *if* $a > b$ *then*
      $a <- a - b$
    *else*
      $b <- b - a$
    *end*
  *end*
  *return* $a$```
))
```

<img width="372" alt="image" src="https://github.com/andreasKroepelin/lovelace/assets/34951714/9920cb37-2a62-4c17-9b8b-31724861a636">

I think that might be better :-)

**Comment by andreasKroepelin** [comment]
Interesting idea! I'm not sure about the name yet but I'd be happy to review a PR implementing this.

**Comment by OrangeX4** [comment]
I have opened a pr: https://github.com/andreasKroepelin/lovelace/pull/3

**Comment by andreasKroepelin** [comment]
Fixed by #3
